upf/io/pddl_writer.py

```diff
@@ -77,16 +79,16 @@ def walk_int_constant(self, expression, args):
         return str(expression.constant_value())
 
     def walk_plus(self, expression, args):
-        assert len(args) == 2
-        return f'(+ {args[0]} {args[1]})'
+        assert len(args) > 1
+        return f'(+ {" ".join(args)})'
```

Comment:
In PDDL `+ - * /` are all binary operators, then if the args are greater than 2 you need to build something like:
`(+ args[0] (+ args[1] ..))`

Reply:
@Framba-Luca tip: you can use the `reduce` function of python: https://docs.python.org/3/library/functools.html#functools.reduce

Reply:
@alvalentini Nice catch!

Reply:
@mikand  Thanks for the suggestion:
I came up with something like this
`return reduce(lambda x, y: f'(+ {y} {x})', iter(args))
`

This should work, right? The test are ok but I guess this isn't really tested yet since the tests are made with binary + and *

Reply:
Yes, it works!

Reply:
@Framba-Luca Yes! (You can omit the `iter()` though): `return reduce(lambda x, y: f'(+ {y} {x})', args)` should work just fine